### DISCUSSION TITLE
Multiscatterplot still available

### DISCUSSION THREAD

**Discussion by roberthardt** [hnolCol/instantclue#22] [Q&A]
Hi,

I am curios if the multiscatterplot from the previous instance of Instant Clue might still be available? So with the plots mirrored by the respective Pearson coefficients.

Best

Robert

**Comment by hnolCol** [comment]
Hi Robert, 

at the moment it is not available. However, I am thinking about two options of how to implement it in the next release:

1. As a specific scatter plot type. Providing an option to: All vs all columns. 
2. As a specific type of the correlation matrix, but instead of the Heatmap, the multi-scatter plot is shown. This would have the benefit of allowing for clustering the columns according to their correlation. The user would then be able to the plot type for both sides for the diagonal for example: scatter - density plot. 

Please let me know what you think. 

Best

Hendrik

**Comment by roberthardt** [comment]
Hi Hendrik,

I would prefer to have some kind of all vs all column plot with the
individual plots mirrored by the Pearson correlations. Something similar to
this R-Plot
http://www.sthda.com/sthda/RDoc/figure/graphs/scatterplot-matrices-psych-package-1.png
.
Moreover It would be cool if one can easily change the order of samples
within this plot.

Best

Robert

Am Mi., 24. März 2021 um 23:16 Uhr schrieb hnolCol ***@***.***

**Reply by hnolCol** [reply]
Hi Robert, 

the more I think about it, I believe it is best to create a new plot type instead of trying to make it work with one of the existing ones (as it was in the old Instant Clue versions). The order of the columns will be in order as drag & dropped into the receiver box and can be changed using the small blue arrows in the top right corner afterwards.

![Screenshot 2021-03-26 at 11 10 52](https://user-images.githubusercontent.com/24191910/112616387-f7f59b00-8e23-11eb-82fa-05ed7a04940e.png)

I will include options to give the Pearson r text a color depending on its values. Another option will be to plot density plot or to display the scatter as a contour plot (e.g. kernel gaussian density) in case of many data points.  I will start to incorporate it now. I will show you a demo here, maybe you can think of additional options that would be required. 

Best wishes and thanks

Hendrik

**Comment by hnolCol** [comment]
Dear Robert, dear all,

I just finished the implementation of the MultiScatter plot. Sadly it took quite long, but I believe that is quite comprehensive now.
There are numerous settings that can be used to tune the appearance of the plot. Find a screenshot below. 

![image](https://user-images.githubusercontent.com/24191910/123101355-41aa4d80-d434-11eb-938d-32d0ff2aed37.png)

Explanation of parameter: 

* multi.scatter.interactive - Turn on/off hover event firing of scatterplots (if there are many plots and datapoint this might lead to high response times)
* multi.scatter.top.right.type - Select plot type of top-right triangle (available scatter, 2D-Histogram) - same with bottom/left
* multi.scatter.diag.type - Diagnal type, you can choose between kde, kde-label or just column name label 
* multi.scatter.sort.columns.by - How to sort the columns within the multi scatter plot. None -> display as entered by drag & drop. Grouping - sort by the current grouping, hierarchical clustering (sort by euclidean/complete based clustering - similar to a correlation matrix.
* ..add.lineregres - add linear regression fit
* ..add.lowess - adds lowess lines
* * ..lowess.frac - Between 0 and 1. The fraction of the data used when estimating each y-value.
* ..lowess.it - The number of residual-based reweightings to perform.
* ..lowess.delta - Distance within which to use linear-interpolation instead of weighted regression.
* ..add.pearns - add Pearson correction coefficient 
* ..background.by.pearson - If enabled, the axis background is color according to the Pearson correlation 
* ..background.colormap - define color map for background coloring
* ..colorScale - Scaling of the colormap (dataMinMax - what ever the min/max Pearson correlation in the data, 0,1 or -1,1 
* ..add.spearman - If enabled, adds spearman correlation to the plot 
* ..add.corr.coeff.to - Select the plot types that should be annotated with the correlation coefficients (pearson, spearman)
* ..kde.kernel - Kernel to calculate the kde (Kernel density estimation) 
* ..kde.bandwith - set the bandwidth for calculation the kde
* ..kde.grid.search - Enable/disable a grid search for the bandwidth 
* ..search.min / search.max - Grid search range
* ..kde.grid.n.cross.val - number of cross validations performed
* ..kde.log.density -  If enabled, the log kde is shown. 


Here is an example of the Iris dataset using the following settings.
![image](https://user-images.githubusercontent.com/24191910/123102213-1d02a580-d435-11eb-822c-28c35f646fbd.png)


Things that need to be implemented for the next release are:

* [ ] Handle x-tick labels  
* [ ] Export to main figure is currently only available for a single chart. 
* [x] Color encoding of scatter  points
* [ ] Size encoding of scatter  points 
* [ ] Style of lowess, lineregress style (color, line width, line style) 
* [ ] Displaying the color code for r values - not clear how to connect that with color encoding.. 


Do you have other suggestions to create such a plot? Any feature you are missing? 

Best wishes

Hendrik
